pkg/resource/template/template_funcs.go

```diff
@@ -42,6 +45,20 @@ func addFuncs(out, in map[string]interface{}) {
 	}
 }
 
+// hashNodeName hashes the node name for ipv6 only systems.
+// This hash is then converted to ipv4 IP to be used as router id.
+func hashNodeName(nodeName string) string {
```

Comment:
Nit: this function is more generic than the comment suggests.  It hashes any string and converts the result to an IPv4 address.  I'd call the function `hashToIPv4` as above and remove the mention of nodeName.

Reply:
ack